test/unit/models/auctions.test.js

```diff
@@ -0,0 +1,113 @@
+const chai = require("chai");
+const sinon = require("sinon");
+const { expect } = chai;
+
+const cleanDb = require("../../utils/cleanDb");
+const firestore = require("../../../utils/firestore");
+const addUser = require("../../utils/addUser");
+
+// Import models
+const auctions = require("../../../models/auctions");
+const walletsQuery = require("../../../models/wallets");
+const auctionModel = firestore.collection("auctions");
+const bidModel = firestore.collection("bids");
+
+// Import fixtures
+const { auctionData } = require("../../fixtures/auctions/auctions");
+const user = require("../../fixtures/user/user")();
+const currencies = require("../../fixtures/currencies/currencies");
+
+const { initial_price: initialPrice, item_type: itemType, end_time: endTime, quantity } = auctionData;
+
+describe("auctions", function () {
+  let seller, auctionId, bidder;
+
+  beforeEach(async function () {
+    seller = await addUser();
+    bidder = await addUser(user[1]);
+    auctionId = await auctions.createNewAuction({ seller, initialPrice, itemType, endTime, quantity });
+  });
+  afterEach(async function () {
+    await cleanDb();
+    sinon.restore();
+  });
+
+  describe("createNewAuction", function () {
+    it("Should create a new auction", async function () {
+      const data = (await auctionModel.doc(auctionId).get()).data();
+
+      expect(data).to.be.a("Object");
+      expect(data.highest_bid).to.be.equal(initialPrice);
+      expect(data.item).to.be.equal(itemType);
+      expect(data.end_time).to.be.equal(endTime);
+      expect(data.quantity).to.be.equal(quantity);
+      expect(data.seller).to.be.equal(seller);
+      expect(data.highest_bidder).to.be.equal(null);
+    });
+  });
+
+  describe("fetchAuctionById", function () {
+    it("Should return the auction data", async function () {
+      const response = await auctions.fetchAuctionById(auctionId);
+
+      expect(response).to.be.a("object");
+      expect(response.highest_bid).to.be.equal(initialPrice);
+      expect(response.item).to.be.equal(itemType);
+      expect(response.end_time).to.be.equal(endTime);
+      expect(response.quantity).to.be.equal(quantity);
+      expect(response.seller).to.be.equal(user[0].username);
+      expect(response.bidders_and_bids).to.be.a("array").with.lengthOf(0);
+      expect(response.highest_bidder).to.be.equal(undefined);
+    });
+    it("Should return not found", async function () {
+      const response = await auctions.fetchAuctionById("undefined");
+
+      expect(response).to.be.equal(false);
```

Comment:
Instead of passing `undefined`, can we pass some other string like `test123`? Someone might think of this an `undefined` keyword of JS and not any random string

Reply:
Ok :+1: